test-basic/write-test.js

```diff
@@ -0,0 +1,818 @@
+                        done();
+                    } catch (e) {
+                        done(e);
+                    }
+                });
+            }, 2000);
+        } catch (e) {
+            done(e);
+        }
+    });
+
+    it('test with multiple objects with one wrong parameter', function (done) {
+        const docsDescriptor = [
+            {uri: '/test/fromDocDescriptors/data1.json', doc: {"desc": "doc1"}},
+            {uri: '/test/fromDocDescriptors/data2.json', doc: {"desc": "doc2"}, collections: ['write']},
+            {
+                uri: '/test/fromDocDescriptors/data3.json',
+                doc: {"desc": "doc3"},
+                collections: ['write', 'doc3'],
+                metadata: {'operation': 'doc3'}
+            }
+        ];
+
+        try {
+            db.rows.execute(op.fromDocDescriptors(docsDescriptor).write(1));
+        } catch (e) {
+            e.toString().includes('Error: doc-cols argument at 0 of PlanModifyPlan.write() must be a PlanDocColsIdentifier value');
+            done();
+        }
+    });
+
+    it('test with multiple objects with two parameters', function (done) {
+        const docsDescriptor = [
+            {uri: '/test/fromDocDescriptors/data1.json', doc: {"desc": "doc1"}},
+            {uri: '/test/fromDocDescriptors/data2.json', doc: {"desc": "doc2"}, collections: ['write']},
+            {
+                uri: '/test/fromDocDescriptors/data3.json',
+                doc: {"desc": "doc3"},
+                collections: ['write', 'doc3'],
+                metadata: {'operation': 'doc3'}
+            }
+        ];
+
+        try {
+            db.rows.execute(op.fromDocDescriptors(docsDescriptor).write(1, null));
+        } catch (e) {
+            e.toString().includes('Error: PlanModifyPlan.write takes a maximum of 1 arguments but received: 2');
+            done();
+        }
+    });
+
+    it('test with single object without parameters', function (done) {
+        const docsDescriptor = {uri: '/test/fromDocDescriptors/data1.json', doc: {"desc": "doc1"}};
+        try {
+            db.rows.execute(op.fromDocDescriptors(docsDescriptor).write());
+            setTimeout(() => {
+                readDocs(['/test/fromDocDescriptors/data1.json', '/test/fromDocDescriptors/data2.json', '/test/fromDocDescriptors/data3.json'], done).then(res => {
+                    try {
+                        const doc = res[0];
+                        doc.collections.length.should.equal(0);
+                        doc.uri.should.equal("/test/fromDocDescriptors/data1.json");
+                        doc.content.desc.should.equal("doc1");
+                        done();
+                    } catch (e) {
+                        done(e);
+                    }
+                });
+            }, 2000);
+        } catch (e) {
+            done(e);
+        }
+    });
+
+    it('test with multiple objects and non-existing columns to write, ignored', function (done) {
```

Comment:
Why does the name here have "ignored" in it?

Reply:
it is not necessary.